src/services/completions.ts

```diff
@@ -1683,6 +1686,15 @@ function createCompletionEntry(
         hasAction = true;
     }
 
+    if (completionKind === CompletionKind.ObjectPropertyDeclaration && contextToken &&
+        findPrecedingToken(contextToken.pos, sourceFile, contextToken)?.kind !== SyntaxKind.CommaToken &&
+        (isMethodDeclaration(contextToken.parent.parent) || isSpreadAssignment(contextToken.parent) || findAncestor(contextToken.parent, (node: Node) => isPropertyAssignment(node))?.getLastToken() === contextToken ||

```

Comment:
1. You don't need to annotate the type of `node,` it'll be inferred.
2. Feed through the `sourceFile` when using `Node` methods so that they don't have to walk back up the tree.

```suggestion
        (isMethodDeclaration(contextToken.parent.parent) || isSpreadAssignment(contextToken.parent) || findAncestor(contextToken.parent, node => isPropertyAssignment(node))?.getLastToken(sourceFile) === contextToken ||
```

Reply:
I'd also rather you just broke this into 2 nested `if`s, where it looks like

```ts
if (completionKind === CompletionKind.ObjectPropertyDeclaration && contextToken && findPrecedingToken(contextToken.pos, sourceFile, contextToken)?.kind !== SyntaxKind.CommaToken) {
    if (isMethodDeclaration(contextToken.parent.parent) ||
        isSpreadAssignment(contextToken.parent) ||
        ...) {
            source = CompletionSource.ObjectLiteralMemberWithComma;
            hasAction = true;
    }
}

Reply:
Wait, you don't need `node => isPropertyAssignment`, just use `isPropertyAssignment` directly.